src/components/ModalBackupWords.js

```diff
@@ -228,87 +257,126 @@ class ModalBackupWords extends React.Component {
       )
     }
 
-    const renderWordsValidation = () => {
-      return this.state.wordsValidation.map((word, idx) => {
-        // The button will only be disabled when all instances of the words have been selected
-        // Might have more than one instance of the same word
-        return (
-          <button key={`${word}${idx}`} className="btn btn-dark mr-2 mt-3" disabled={hathorLib.helpers.elementCount(this.state.chosenWords, word) === hathorLib.helpers.elementCount(this.state.wordsValidation, word)} onClick={(e) => {this.validationWordClicked(word)}}>{word}</button>
-        )
-      });
-    }
-
-    const renderChosenWords = () => {
-      return this.state.chosenWords.map((word, idx) => {
-        return (
-          <div key={`${word}${idx}`}>
-            <span className="word-chosen">{word}</span>
-            <i className="fa fa-close" onClick={(e) => {this.chosenWordRemoved(word)}} />
-          </div>
-        )
-      });
-    }
+    const renderValidationStep = (validationStep) => {
+      return (
+        <article className='w-100'>
+          <h2 className='validation-step-index d-flex flex-row justify-content-center btn-block mb-16'>
+            {validationStep.backupIndex}
+          </h2>
+          <section
+            className='validation-step-words btn-group d-flex flex-row justify-content-center'
+            role='list'
+            aria-label='Backup Validation Options'
+          >
+            {validationStep.options.map((option) => (
+              <div key={option} role='listitem'>
+                <button
+                  className={`btn btn-dark btn-block validate-step-option ${
+                    validationStep.chosenOption === option
+                      ? 'chosen-option'
+                      : ''
+                  }`}
+                  onClick={() =>
+                    this.handleValidationStepOption(option, validationStep)
+                  }
+                  disabled={this.state.errorMessage}
+                >
+                  {option}
+                </button>
+              </div>
+            ))}
+          </section>
+        </article>
+      );
+    };
 
-    const renderValidateBackup = () => {
+    const renderValidationProgress = () => {
       return (
-        <div>
-          <p>{t`Please, select the words below in the correct order. This is for your safety and to ensure you wrote them down correctly.`}</p>
-          <div className="chosen-words-wrapper w-100 d-flex flex-row flex-wrap align-items-start">
-            {renderChosenWords()}
-          </div>
-          <div className="words-validation d-flex flex-row align-items-start flex-wrap">
-            {renderWordsValidation()}
-          </div>
+        <div className='validation-progress d-flex flex-row justify-content-center'>
+          {this.state.validationSteps.map((validationStep) => (
+            <div
+              key={validationStep.correctWord}
```

Comment:
This can be a problem if we have repeated words and both are chosen in the validation. Better to use the list index, no?

Reply:
You are right. I will use the list index, although it is generally not [recommended](https://reactjs.org/docs/lists-and-keys.html), it is better than `validationStep.correctWord`, thanks!